ddtrace/writer.py

```diff
@@ -112,27 +114,51 @@ def _on_shutdown(self):
                     time.sleep(0.05)
 
     def _target(self):
+        result_traces = None
+        result_services = None
+
         while True:
             traces = self._trace_queue.pop()
             if traces:
                 # If we have data, let's try to send it.
                 try:
-                    self.api.send_traces(traces)
+                    result_traces = self.api.send_traces(traces)
                 except Exception as err:
                     log.error("cannot send spans: {0}".format(err))
 
             services = self._service_queue.pop()
             if services:
                 try:
-                    self.api.send_services(services)
+                    result_services = self.api.send_services(services)
                 except Exception as err:
                     log.error("cannot send services: {0}".format(err))
 
             elif self._trace_queue.closed():
-                # no traces and the queue is closed. our work is done.
+                # no traces and the queue is closed. our work is done
                 return
 
-            time.sleep(1) # replace with a blocking pop.
+            log_level = log.debug
+            if result_traces and getattr(result_traces, "status", None) >= 400:
```

Comment:
this code is the same as the block below... you can merge the two blocks or use a separated function like `_log_error_status(...)`